torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2676,6 +2676,29 @@ def _make_tensor_helper(shape, low=None, high=None):
                                 dtypes=[torch.bfloat16])),
                    safe_casts_outputs=True,
                    handles_complex_extremals=False),
+    UnaryUfuncInfo('square',
+                   ref=np.square,
+                   dtypes=all_types_and_complex_and(torch.bool),
+                   dtypesIfCPU=all_types_and_complex_and(torch.bool),
+                   dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
+                   decorators=(precisionOverride({torch.complex64: 3e-4, torch.bfloat16: 3e-1}),),
+                   skips=(
+                       # Reference: https://github.com/pytorch/pytorch/issues/52549
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_hard',
+                                dtypes=[torch.cfloat, torch.cdouble]),
+                       # >>> t = torch.tensor(complex(-0.01, float("inf")))
+                       # >>> np.square(t.numpy())
+                       # (-inf-infj)
+                       # >>> t.square()
+                       # tensor(-inf-infj)
+                       # >>> t.cuda().square()
+                       # tensor(inf+nanj, device='cuda:0')
```

Comment:
Is above skip worth a separate issue?

Reply:
A skip for pow's handling of complex extremals would be helpful if one doesn't already exist.

Reply:
I couldn't find anything similar for `pow`. Have added a very basic test for the same (which should fail once this is fixed).

Reply:
Hello, there was some comment about this op not being enabled for half types (float16, bfloat16) due to lack of support for `pow`, but I can't find it anymore. :(

Reply:
This PR https://github.com/pytorch/pytorch/pull/53255 restricted the variant testing to float dtype. Hence the skips for variant testing for float16, bfloat16 are not necessary. 

Also note that it adds 
> Adds backwards supported testing to clarify failures that were coming from variant testing

(so maybe these should have had failed :thinking: )

Reply:
I had asked because I thought I had read somewhere that I should add `float16` & 'bfloat16` to the `dtypes` list when `pow` for these dtypes would be enabled on CPU.

But now I discovered that `test_unsupported_dtypes` would fail for `square` for these dtypes on CPU if these dtypes are not added to the OpInfo `dtypes` list, after `pow` is enabled for these dtypes on CPU.